src/index.ts

```diff
@@ -225,11 +222,11 @@ export class Upload extends Pumpify {
   createURI(callback: CreateUriCallback) {
     const metadata = this.metadata;
 
-    const reqOpts: RequestOptions = {
+    const reqOpts: r.OptionsWithUrl = {
       method: 'POST',
       url: [BASE_URI, this.bucket, 'o'].join('/'),
-      params: {name: this.file, uploadType: 'resumable'},
-      data: metadata,
+      qs: {name: this.file, uploadType: 'resumable'},
+      body: metadata,
```

Comment:
I think we want `json` here instead of `body`:

> `json` - sets body to JSON representation of value and adds Content-type: application/json header. Additionally, parses the response body as JSON.
> \- https://github.com/request/request